packages/cdk-ecs-alb/lib/index.ts

```diff
@@ -0,0 +1,84 @@
+import * as cdk from "@aws-cdk/core";
+import * as ecspattern from "@aws-cdk/aws-ecs-patterns";
+import * as ecs from "@aws-cdk/aws-ecs";
+import * as ecr from "@aws-cdk/aws-ecr";
+import * as ec2 from "@aws-cdk/aws-ec2";
+import * as logs from "@aws-cdk/aws-logs";
+import { Duration } from "@aws-cdk/core";
+
+export interface CdkEcsAlbProps {
+    clusterName: string;
+    appName: string;
+    vpcId: string;
+    securityGroupId: string;
+    repositoryName: string;
+}
+
+export class CdkEcsAlb extends cdk.Construct {
+    constructor(scope: cdk.Construct, id: string, props: CdkEcsAlbProps) {
+        super(scope, id);
+
+        const vpc = ec2.Vpc.fromLookup(this, "VPC", { vpcId: props.vpcId });
+
+        const repository = ecr.Repository.fromRepositoryName(
+            this,
+            "Repo",
+            props.repositoryName
+        );
+
+        const securityGroup = ec2.SecurityGroup.fromLookup(
+            this,
+            "Security Group",
+            props.securityGroupId
+        );
+
+        const cluster = ecs.Cluster.fromClusterAttributes(this, "Cluster", {
+            clusterName: props.clusterName,
+            vpc: vpc,
+            securityGroups: [securityGroup],
+        });
+
+        const taskDefinition = new ecs.Ec2TaskDefinition(
+            this,
+            "TaskDefinition",
+            {
+                networkMode: ecs.NetworkMode.NAT,
+            }
+        );
+
+        taskDefinition.addContainer("Container", {
+            image: ecs.ContainerImage.fromEcrRepository(repository),
+            memoryLimitMiB: 4096,
+            cpu: 2048,
+            portMappings: [
+                {
+                    containerPort: 80,
+                    hostPort: 0,
+                    protocol: ecs.Protocol.TCP,
+                },
+                {
+                    containerPort: 443,
+                    hostPort: 0,
+                    protocol: ecs.Protocol.TCP,
+                },
+            ],
+            entryPoint: ["powershell", "-Command"],
+            command: ["C:\\ServiceMonitor.exe w3svc"],
+            logging: ecs.LogDriver.awsLogs({
+                logGroup: new logs.LogGroup(this, "LogGroup", {
+                    logGroupName: `${props.appName}-ecs`,
+                }),
+                streamPrefix: "ecs",
+            }),
+        });
+
+        new ecspattern.ApplicationLoadBalancedEc2Service(this, "ALB", {
```

Comment:
I wouldn't name this construct "ALB" since this includes way more than just the load balancer.